### PR TITLE
Civ 845 feature ras api call

### PR DESCRIPTION
**Pull request by osheaf01**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-845

### Change description ###
Added RoleAssignmentApi to make an /am/role-assignment/actor/{actorId} call to RAS
Added RoleAssignmentService to expose a method to get role assignments for an actorId
Added TestUserConfiguration for a test user
Exposed a getCredentials method in the new CasesController class
Added a simple test in CasesControllerTest.

Have tested locally by running http://localhost:4000/claims/actors/1d70e58a-73ab-4423-83b0-9cd82a810250 and verifying it returns a Role Assignment JSON

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/ras/client/RoleAssignmentsApi.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        value = "/am/role-assignments/actors/{actorId}",
+        consumes = APPLICATION_JSON_VALUE,
+        headers = CONTENT_TYPE + "=" + APPLICATION_JSON_VALUE
+    )
+    String getRoleAssignments(
+        @RequestHeader(HttpHeaders.AUTHORIZATION) String authorization,
+        @RequestHeader(SERVICE_AUTHORIZATION) String serviceAuthorization,
+        @PathVariable("actorId") String actorId);
+
+}
```
Comment (LeonardoPalmeiro) [inline src/main/java/uk/gov/hmcts/reform/civil/ras/client/RoleAssignmentsApi.java:28]: there is no test in this part of the code We have to create it

File: src/main/java/uk/gov/hmcts/reform/civil/service/RoleAssignmentsService.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    public String getRoleAssignments() {
+        return this.getRoleAssignments(getActorId());
+    }
+
+
+    private String getActorId() {
+        return idamClient.getUserDetails(getUserToken()).getId();
+    }
+
+    private String getUserToken() {
```
[the remark this answers is not in the document]
Comment (osheaf01) [reply src/main/java/uk/gov/hmcts/reform/civil/service/RoleAssignmentsService.java:43]: Don't think I need to unit test private methods?

File: src/main/java/uk/gov/hmcts/reform/civil/config/TestUserConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package uk.gov.hmcts.reform.civil.config;
+
+import lombok.Data;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+
+@Data
+@Configuration
+public class TestUserConfiguration {
```
[the remark this answers is not in the document]
Comment (osheaf01) [reply src/main/java/uk/gov/hmcts/reform/civil/config/TestUserConfiguration.java:9]: It's in the same folder as all the other Configuration classes?

File: bin/utils/service.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+## Options:
+##    - microservice_name: Name of the microservice. Default to `ccd_gw`.
+##
+## Returns a valid IDAM service token for the given microservice.
+
+microservice="${1:-civil}"
+curl -X POST \
+  -H "Content-Type: application/json" \
+  -d '{"microservice":"'${microservice}'"}' \
+  http://localhost:4502/testing-support/lease
```
[the remark this answers is not in the document]
Comment (osheaf01) [reply bin/utils/service.sh:13]: Going to delete this

File: src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/cases/CasesControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    @SneakyThrows
+    public void shouldReturnOk() {
+        doGet(BEARER_TOKEN, CLAIMS_LIST_URL)
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    @SneakyThrows
+    public void shouldReturnRASAssignment() {
+        when(roleAssignmentsService.getRoleAssignments(ACTOR_ID)).thenReturn(ACTOR_ROLE);
```
[the remark this answers is not in the document]
Comment (osheaf01) [reply src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/cases/CasesControllerTest.java:56]: Have pushed a code change to fix this.

File: src/main/java/uk/gov/hmcts/reform/civil/config/TestUserConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+    private final String username;
+    private final String password;
+
+    public TestUserConfiguration(@Value("${civil.test-user.username}") String username,
```
[the remark this answers is not in the document]
Comment (osheaf01) [reply src/main/java/uk/gov/hmcts/reform/civil/config/TestUserConfiguration.java:14]: The @Value there is a Spring annotation, not lombok. 
I just used the model for the other UserConfiguration classes in the same package.

File: src/main/java/uk/gov/hmcts/reform/civil/ras/client/RoleAssignmentsApi.java (hunk 1 above)
Comment (osheaf01) [reply src/main/java/uk/gov/hmcts/reform/civil/ras/client/RoleAssignmentsApi.java:28]: It's an interface, it's called by RoleAssignmentsService, which is tested in CasesControllerTest.

File: src/main/java/uk/gov/hmcts/reform/civil/config/TestUserConfiguration.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (osheaf01) [reply src/main/java/uk/gov/hmcts/reform/civil/config/TestUserConfiguration.java:9]: Think we can resolve this?

**Review by sampankumar — APPROVED** [review]

**Review by LeonardoPalmeiro — APPROVED** [review]

**Review by laarmada — COMMENTED** [review]
seems ok

**Pull request merged**